onnx/defs/tensor/defs.cc

```diff
@@ -1655,8 +1646,12 @@ ONNX_OPERATOR_SET_SCHEMA(
                 return axis < 0 ? axis + input_ndim : axis;
               });
 
-          for (int i = 0, j = 0; i < input_ndim; ++i) {
-            if (std::find(axes.begin(), axes.end(), i) != axes.end()) {
+          for (int i = 0; i < input_ndim; ++i) {
```

Comment:
if-then-else may need to be refactored. If axes is not specified, and input_shape.dim(i) has a symbolic value, it will fall into the else branch, which seems wrong.

Reply:
The case where (not_specify_axes) is a bit more involved. If there is even one unknown input-dimension, we can't really set any of the other dimensions, since the output rank itself becomes unknown.